projects/smoke_pca9539r/src/main.c

```diff
@@ -0,0 +1,120 @@
+// Smoke test for pca9539r gpio expander
+// Checks all pins configure to inputs/outputs, then toggles gpio states of all 16 pins
+// At every stage, prints comparison of configuration registers against expected results
+// I2C settings may need to be configured to match wiring on board
+// Changing WAIT_TIME_MILLIS allows different times between toggling
+
+#include "gpio.h"
+#include "i2c.h"
+#include "interrupt.h"
+#include "log.h"
+#include "pca9539r_gpio_expander.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+// used to adjust time between gpio pin toggling
+#define WAIT_TIME_MILLIS 1000
+
+#define PCA9539_I2C_ADDRESS 0x74  // PCA9539 address
+#define I2C_PORT I2C_PORT_2
+// I2C_PORT_1 has SDA on PB9 and SCL on PB8
+// I2C_PORT_2 has SDA on PB11 and SCL on PB10
+
+#define PIN_I2C_SCL \
+  { GPIO_PORT_B, 10 }
+#define PIN_I2C_SDA \
+  { GPIO_PORT_B, 11 }
+
+void setup_test(void) {
+  I2CSettings i2c_settings = {
+    .speed = I2C_SPEED_FAST,
+    .sda = PIN_I2C_SDA,
+    .scl = PIN_I2C_SCL,
+  };
+  i2c_init(I2C_PORT, &i2c_settings);
+  pca9539r_gpio_init(I2C_PORT, PCA9539_I2C_ADDRESS);
+}
+
+// initialize all pins to in/out - must be called after pca9539r_gpio_init
+StatusCode pca9539r_init_all_pins(Pca9539rGpioDirection direction) {
```

Comment:
A final nit I didn't catch earlier: this should be `static` and called `prv_pca9539r_init_all_pins`